src/lib/version/version.c

```diff
@@ -58,6 +58,27 @@ enum FIRMWARE_TYPE {
 	FIRMWARE_TYPE_RELEASE = 255
 };
 
+typedef enum {
+	PATCH=0,
+	MINOR=8,
+	MAJOR=16
+} version_type_t;
+
+static void string_to_int(int8_t mag, version_type_t v_type, uint32_t *ver, int start, const char *tag)
```

Comment:
Please simplify this method:
- the name `mag` is misleading and has incorrect type: it describes the parsing end and should be int
- simplify the signature so that the function does what it says. I suggest: `uint32_t string_to_int(const char *tag, int start, int end);`, and move `*ver = (temp_ver << v_type) + *ver;` to the caller.
- add API documentation (see other methods)